thirdeye-ui/src/app/components/alert-composite-filters-modal/alert-composite-filters-modal.component.tsx

```diff
@@ -0,0 +1,231 @@
+            ).map((item) => item.params.queryFilters);
+
+            // Parse the stringified `queryFilter=` to an object
+            // `dimensionFilters` format for precise comparison
+            const selectedDimensionFilters = selectedEnumerationItemsInAlert
+                .map((query) =>
+                    query
+                        .split(NAME_JOIN_KEY)
+                        .map((dim) => dim.split("="))
+                        .filter((v) => v.filter((g) => g.length > 0))
+                        .filter((v) => v.length > 0 && v[0].length > 0)
+                        .map(([k, v]) => ({ [k]: v.slice(1, -1) }))
+                )
+                .map((entries) => Object.assign({}, ...entries));
+
+            // Get the table rows from the cohorts list
+            const cohortDimensionsTableRow = cohortsResponse?.results
+                ? cohortsResponse?.results.map(getCohortTableRowFromData)
+                : [];
+
+            // Filter for the cohorts that match any enumeration item in alert
+            const cohortsInAlert = cohortDimensionsTableRow.filter(
+                (dimension) =>
+                    selectedDimensionFilters.some((selected) =>
+                        isEqual(selected, dimension.dimensionFilters)
+                    )
+            );
+
+            return cohortsInAlert;
+        }, [alert, status, cohortsResponse]);
+
+        const handleCreateBtnClick = (): void => {
+            const enumerationItemConfiguration = selectedCohorts.map(
+                (cohort) => {
+                    const criteria = Object.keys(cohort.dimensionFilters).map(
+                        (k) => {
+                            return `${k}='${cohort.dimensionFilters[k]}'`;
+                        }
+                    );
+                    const joined = criteria.join(" AND ");
+
+                    return {
+                        params: {
+                            queryFilters: ` AND ${joined}`,
+                        },
+                    };
+                }
+            );
+
+            let queryFilters = "${queryFilters}";
+
+            // Append the custom query filter if exists
+            if (queryValue) {
+                queryFilters = `${queryFilters} AND ${queryValue}`;
+            }
+
+            onAlertPropertyChange({
+                templateProperties: {
+                    ...alert.templateProperties,
+                    queryFilters,
+                    enumerationItems: enumerationItemConfiguration,
+                },
+            });
+            onUpdateCompositeFiltersChange({
+                ...alert.templateProperties,
+                queryFilters,
+                enumerationItems: enumerationItemConfiguration,
+            });
+            onCancel();
+        };
+
+        return (
+            <Modal
+                initiallyOpen
+                maxWidth="md"
+                title={t("label.add-dimensions")}
+                trigger={() => <></>}
```

Comment:
Nit: When cases like these pop-up, it can be helpful in the long run to modify the common component (`Modal` in this case) to have `trigger` as an optional parameter.

Reply:
@alcatraz627 I have updated the Modal component to have trigger as an optional parameter. Please check.